pkg/event/events.go

```diff
@@ -50,6 +50,9 @@ func getPolicyKind(policy kyvernov1.PolicyInterface) string {
 }
 
 func NewPolicyAppliedEvent(source Source, engineResponse *response.EngineResponse) Info {
+	if engineResponse.Policy.GetSpec().IsEmitAppliedEvents() {
+		return Info{}
+	}
```

Comment:
@eddycharly I had to return Empty info object here, is that correct?

Reply:
Sounds wrong, why call `NewPolicyAppliedEvent` in this case ?

Reply:
y thought process was, if we won't emit policy applied events, then don't generate them if the flag is set

Reply:
The flag should be checked by the caller IMHO, not in `NewPolicyAppliedEvent`.